src/main/java/org/walkmod/javalang/compiler/types/TypesLoaderVisitor.java

```diff
@@ -135,22 +135,27 @@ public TypesLoaderVisitor(SymbolTable symbolTable, SymbolActionProvider actionPr
     * @param imported
     *           if it appears as an import declaration. It is important, because the simple name is
     *           the part after the $ symbol if it is an inner class.
+    * @param importedName
+    *           the name of the import to resolve nested classes against (optional)
     * @return the simple name to be pushed into the symbol table
     */
-   private String getSymbolName(String name, boolean imported) {
+   //@VisibleForTesting
+   static String registrationSymbolName(final String name, final boolean imported, String importedName) {
       int index = name.lastIndexOf(".");
       String simpleName = name;
       if (index != -1) {
          simpleName = name.substring(index + 1);
       }
-      if (imported) {
-         index = simpleName.lastIndexOf("$");
-         if (index != -1) {
-            simpleName = simpleName.substring(index + 1);
+      if (importedName != null && name.startsWith(importedName + "$")) {
```

Comment:
I am not sure about consequnences and just added what was missing, fixing an obvious case because of no test cases and not enough overview about what should be in the symbol table. The fact that this case got undetected so far and I wasted 2 days for hunting it in the semantic code part made be nervous to change too much.